lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Context/ContextRuntime.cs

```diff
@@ -68,7 +84,28 @@ internal sealed class ContextRuntime : IDisposable
                 Optional<ContextRuntime> parentContext)
         {
             _serviceInjector = serviceInjector;
-            _injectedServices = Optional<ISet<object>>.Of(serviceInjector.GetNamedInstance<ServicesSet, ISet<object>>());
+
+            // Note that for Service objects and handlers, we are not merging them into a separate
+            // class (e.g. ServiceContainer) due to the inability to allow service stacking if an instance 
+            // of such a class were to be materialized. i.e. if a ServiceContainer object were initialized
+            // and a child ServiceConfiguration is submitted, when the child service injector tries to
+            // get the relevant handlers and services set, it will get the same set of handlers as
+            // previously instantiated by the parent injector, and thus will not allow the stacking
+            // of ServiceConfigurations.
+            _injectedServices = serviceInjector.GetNamedInstance<ServicesSet, ISet<object>>();
+
+            _serviceContextStartHandlers = 
+                serviceInjector.GetNamedInstance<ContextConfigurationOptions.StartHandlers, ISet<IObserver<IContextStart>>>();
+
+            _serviceContextStopHandlers = 
+                serviceInjector.GetNamedInstance<ContextConfigurationOptions.StopHandlers, ISet<IObserver<IContextStop>>>();
+
+            _serviceTaskStartHandlers = 
```

Comment:
They are invoked when Contexts start on the Service. They form the basis of the set of ContextHandlers that are merged with those specified in ContextConfiguration.